src/setfit/trainer.py

```diff
@@ -363,19 +361,17 @@ def train(
 
                 train_dataloader = DataLoader(train_examples, shuffle=True, batch_size=batch_size)
                 train_loss = self.loss_class(self.model.model_body)
-                train_steps = len(train_dataloader) * num_epochs
 
             logger.info("***** Running training *****")
             logger.info(f"  Num examples = {len(train_examples)}")
             logger.info(f"  Num epochs = {num_epochs}")
-            logger.info(f"  Total optimization steps = {train_steps}")
+            logger.info(f"  Total optimization steps = {len(train_dataloader) * num_epochs}")
             logger.info(f"  Total train batch size = {batch_size}")
 
             warmup_steps = math.ceil(train_steps * self.warmup_proportion)
```

Comment:
```suggestion
            warmup_steps = math.ceil(train_steps * self.warmup_proportion)
```
This still uses the `train_steps` variable.